### PR TITLE
Error handling infra

### PR DESCRIPTION
**Pull request by yhaspel**
In this commit:
initial infra for error handling in app tabs in Main.js

Screen Shot 2019-06-17 at 3 40 40 PM

### CODE REVIEW

File: app/components/errorHandler/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    if (typeof onError === 'function') {
+      try {
+        onError.call(this, error, info ? info.componentStack : '');
+      } catch (ignoredError) {
+        // ignore this error
+      }
+    }
+
+    this.setState({ error, info });
```
Comment (IlyaVi) [inline app/components/errorHandler/ErrorBoundary.js]: refresh action should be done by setting error to null once again

File: app/components/errorHandler/ErrorBoundary.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  render() {
+    const { error, info } = this.state;
+    const { children, FallbackComponent, mode, onButtonPress } = this.props;
+
+    if (error !== null) {
+      return <FallbackComponent componentStack={info ? info.componentStack : ''} error={error} mode={mode} onButtonPress={onButtonPress} />;
+    }
+    return children || null;
```
Comment (IlyaVi) [inline app/components/errorHandler/ErrorBoundary.js]: remove null

File: app/components/errorHandler/ErrorBoundary.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+type ErrorInfo = {
+  componentStack: string
+};
+
+type Props = {
+  children?: any,
+  FallbackComponent?: ComponentType<any>,
+  onError?: ({ error: Error, componentStack: string, afterErrorAction: 'refresh' | 'retry' | 'go back' }) => void,
```
Comment (IlyaVi) [inline app/components/errorHandler/ErrorBoundary.js]: remove

File: app/components/errorHandler/ErrorBoundary.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  canGoBack?: boolean
+};
+
+type ErrorInfo = {
+  componentStack: string
+};
+
+type Props = {
+  children?: any,
+  FallbackComponent?: ComponentType<any>,
```
Comment (IlyaVi) [inline app/components/errorHandler/ErrorBoundary.js]: remove

File: app/components/errorHandler/ErrorBoundary.js (hunk 1 above)
Comment (yhaspel) [reply app/components/errorHandler/ErrorBoundary.js]: ok

File: app/components/errorHandler/ErrorHandlerModal.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      <Wrapper>
+        <Text>{error.message || 'Error'}</Text>
+        <ErrorDetails>
+          <DetailsText>{componentStack}</DetailsText>
+        </ErrorDetails>
+        <ButtonsWrapper>
+          <SmButton
+            text="Refresh"
+            theme="orange"
+            onPress={(e) => {
```
Comment (IlyaVi) [inline app/components/errorHandler/ErrorHandlerModal.js]: export this to standalone handler

Comment (yhaspel) [reply app/components/errorHandler/ErrorHandlerModal.js]: ok

**Pull request merged**